lib/grape/jsonapi/services/content_negotiator.rb

```diff
@@ -0,0 +1,41 @@
+module Grape
+  module Jsonapi
+    module Services
+      class ContentNegotiator
+        VALID_MEDIA_TYPE = 'application/vnd+json'.freeze
+
+        def initialize(headers)
+          @content_type   = headers['Content-Type']
+          @accept_header  = headers['Accept']
+        end
+
+        def self.run(*args)
+          new(*args).run
+        end
+
+        def run
+          raise unsupported_media_type_error unless valid?(content_type)
+          raise not_acceptable_error unless valid?(accept_header)
+
+          true
+        end
+
+        private
+
+        attr_reader :content_type, :accept_header
+
+        def valid?(header)
+          VALID_MEDIA_TYPE == header || header.nil?
```

Comment:
Valid accept header is something like `application/vnd.geoffrey-v20170505+json`, no ?

Reply:
It's not JSON API compliant, so no. Instead of versioning in the Accept header we're now using the Accept-Version header.

Reply:
According to [this](https://github.com/alexshook/grape-jsonapi_entity/blob/11cfab8fd05de10a699c685f5c3adab69ac1743e/lib/grape/jsonapi/services/content_negotiator.rb#L9) it looks like we're pulling the `Accept` header, not `Accept-Version`.  Also, tests or it didn't happen lol ;)